cephfs/directory.go

```diff
@@ -108,6 +122,15 @@ func toDirEntry(de *C.struct_dirent) *DirEntry {
 	}
 }
 
+// toDirEntryPlus converts c structs set by ceph_readdirplus_r to our go
+// wrapper.
+func toDirEntryPlus(de *C.struct_dirent, s C.struct_ceph_statx) *DirEntryPlus {
+	return &DirEntryPlus{
+		DirEntry: *toDirEntry(de),
+		statx:    cStructToCephStatx(s),
```

Comment:
Well, I did think it was kinda ugly but not worth wringing my hands over. DirEntryPlus embeds dir entry so to construct it we need a DirEntry not a pointer to DirEntry, thus the de-reference.
It might actually be a good idea to have cStructToCephStatx use pointers to avoid any unwanted copying up until we actually make a CephStatx and copy the values out of the C struct.

I think I can probably clean this up a bit in a follow up PR since none of the functionality/behavior will be affected.